easypermissions/src/main/java/pub/devrel/easypermissions/AppSettingsDialog.java

```diff
@@ -65,35 +61,37 @@ private AppSettingsDialog(Parcel in) {
     }
 
     private AppSettingsDialog(@NonNull final Object activityOrFragment,
-                              @NonNull final Context context,
                               @StyleRes int themeResId,
                               @Nullable String rationale,
                               @Nullable String title,
                               @Nullable String positiveButtonText,
                               @Nullable String negativeButtonText,
-                              @Nullable DialogInterface.OnClickListener negativeListener,
                               int requestCode) {
         mActivityOrFragment = activityOrFragment;
-        mContext = context;
         mThemeResId = themeResId;
         mRationale = rationale;
         mTitle = title;
         mPositiveButtonText = positiveButtonText;
         mNegativeButtonText = negativeButtonText;
-        mNegativeListener = negativeListener;
         mRequestCode = requestCode;
     }
 
-    void setActivityOrFragment(Object activityOrFragment) {
-        mActivityOrFragment = activityOrFragment;
-    }
-
-    void setContext(Context context) {
-        mContext = context;
+    static AppSettingsDialog parse(Intent intent, Activity activity) {
+        AppSettingsDialog dialog = intent.getParcelableExtra(AppSettingsDialog.EXTRA_APP_SETTINGS);
+        dialog.mActivityOrFragment = activity;
+        return dialog;
     }
 
-    void setNegativeListener(DialogInterface.OnClickListener negativeListener) {
-        mNegativeListener = negativeListener;
+    private static Context parseContext(Object activityOrFragment) {
```

Comment:
Could we make this method something we call just once (at `build()` time for example?) and then save the `Context` in a variable rather than re-casting `activityOrFragment` every time we want to use it?